mod-source-record-manager-server/src/main/java/org/folio/services/progress/JobExecutionProgressServiceImpl.java

```diff
@@ -26,11 +30,24 @@
 
   @Override
   public Future<JobExecutionProgress> initializeJobExecutionProgress(String jobExecutionId, Integer totalRecords, String tenantId) {
-    return jobExecutionProgressDao.initializeJobExecutionProgress(jobExecutionId, totalRecords, tenantId);
+    return jobExecutionProgressDao.initializeJobExecutionProgress(jobExecutionId, totalRecords, tenantId)
+      .compose(jobExecutionProgress -> {
+        if (jobExecutionProgress == null) {
+          return Future.succeededFuture();
+        }
+        JobMonitoring jobMonitoring = new JobMonitoring();
+        jobMonitoring.setJobExecutionId(jobExecutionProgress.getJobExecutionId());
+        jobMonitoring.setLastEventTimestamp(new Date());
+        jobMonitoring.setNotificationSent(false);
+        return jobMonitoringService.save(jobMonitoring, tenantId)
```

Comment:
```suggestion
         return jobMonitoringService.saveNew(jobExecutionProgress.getJobExecutionId(), tenantId)
```
I think you can move the logic of entity creation to the service.